internal/gatewayapi/translator.go

```diff
@@ -273,28 +285,71 @@ func (t *Translator) Translate(resources *resource.Resources) (*TranslateResult,
 func (t *Translator) GetRelevantGateways(resources *resource.Resources) (
 	acceptedGateways, failedGateways []*GatewayContext,
 ) {
+	// make sure that the EnvoyProxyForGatewayClass is valid
+	if ep := resources.EnvoyProxyForGatewayClass; ep != nil {
+		err := validateEnvoyProxy(ep)
+		if err != nil {
+			t.Logger.Error(err, "Skipping GatewayClass because EnvoyProxy is invalid",
+				"gatewayclass", t.GatewayClassName,
+				"envoyproxy", ep.Name, "namespace", ep.Namespace)
+			status.SetGatewayClassAccepted(resources.GatewayClass,
+				false, string(gwapiv1.GatewayClassReasonInvalidParameters),
+				fmt.Sprintf("%s: %v", status.MsgGatewayClassInvalidParams, err))
+			return
+		}
+	}
+
 	for _, gateway := range resources.Gateways {
 		if gateway == nil {
 			panic("received nil gateway")
 		}
 
-		if gateway.Spec.GatewayClassName == t.GatewayClassName {
-			gc := &GatewayContext{
-				Gateway: gateway,
-			}
+		logKeysAndValues := []any{
+			"namespace", gateway.Namespace, "name", gateway.Name,
+		}
+		if gateway.Spec.GatewayClassName != t.GatewayClassName {
+			t.Logger.Info("Skipping Gateway because GatewayClassName doesn't match", logKeysAndValues...)
+		}
+
+		gCtx := &GatewayContext{
```

Comment:
previously this was only being added when `gateway.Spec.GatewayClassName == t.GatewayClassName `

Reply:
so I missed continue after L311?

Reply:
looks like it

Reply:
fixed.